src/commands/config/create.ts

```diff
@@ -0,0 +1,191 @@
+import { readFile } from "../../implementations/utils/disk";
+import { create as createWallet } from "../../implementations/wallet/create";
+import { getLogger } from "../../logger";
+import { highlight } from "../../utils";
+import { handler as createTemporaryDns } from "../dns/txt-record/create";
+import { CreateConfigCommand } from "./config.type";
+
+interface ConfigFile {
+  wallet: string;
+  forms: Form[];
+}
+
+interface Form {
+  type: "VERIFIABLE_DOCUMENT" | "TRANSFERABLE_RECORD";
+  defaults: {
+    issuers: OpenAttestationDocument[];
+  };
+}
+
+const { trace } = getLogger("config:create");
+
+export const command = "create [options]";
+
+export const describe = "Create a config file";
+
+export const builder = (yargs: Argv): Argv =>
+  yargs
+    .option("output-dir", {
+      alias: "od",
+      description: "Write output to a directory",
+      type: "string",
+      demandOption: true,
+    })
+    .option("encrypted-wallet-path", {
+      type: "string",
+      description: "Path to file containing wallet.json",
+      normalize: true,
+    })
+    .option("config-template-path", {
+      type: "string",
+      description: "Path to file containing config template",
+      normalize: true,
+    })
+    .option("config-type", {
+      type: "string",
+      description: "type of config to create (i.e. tradetrust)",
+      normalize: true,
+      choices: ["tradetrust"],
+    });
+
+export const handler = async (args: CreateConfigCommand): Promise<void> => {
+  trace(`Args: ${JSON.stringify(args, null, 2)}`);
+  if (!fs.existsSync(args.outputDir)) {
+    fs.mkdirSync(args.outputDir);
+  }
+
+  try {
+    let walletPath = "";
+    if (!args.encryptedWalletPath) {
+      info(`Please enter password to create wallet`);
+      const createWalletParams = {
+        outputFile: `${args.outputDir}`,
+        fund: "ropsten",
+      };
+      walletPath = await createWallet(createWalletParams);
+      info(`Wallet created at ${walletPath}`);
+    }
+    const walletFilePath = walletPath ? walletPath : args.encryptedWalletPath;
+    const wallet = await readFile(walletFilePath);
+    const walletObject = JSON.parse(wallet);
+
+    const configFile: ConfigFile = JSON.parse(
+      await readFile(
+        args.configTemplatePath ? args.configTemplatePath : "src/commands/config/__tests__/initial-config.json"
+      )
+    );
+    configFile.wallet = wallet;
+
+    const formsInTemplate = configFile.forms;
+    let documentStoreAddress = "";
+    let tokenRegistryAddress = "";
+    let verifiableDocumentDnsTxtName = "";
+    let verifiableDocumentDnsDidName = "";
+    let tokenRegistryDnsName = "";
+
+    const documentStoreTemporaryDnsParams = {
+      networkId: 3,
+      address: "",
+      publicKey: "",
+      sandboxEndpoint: "https://sandbox.openattestation.com",
+    };
+
+    // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+    const verifiableDnsTxt = async () => {
```

Comment:
let's create function that retusn value (and do one thing), instead of function that do multiple things (deploy doc store + deploy dns txt) and update global variables

Reply:
ok

Reply:
not addressed, `createTemporaryDnsWithDocumentStore` for instance still update a scoped variable

Reply:
to make it clearer, remove the following, and make it work

```ts
    let documentStoreAddress = "";
    let tokenRegistryAddress = "";
    let verifiableDocumentDnsTxtName = "";
    let verifiableDocumentDnsDidName = "";
    let tokenRegistryDnsName = "";
```

Reply:
about this, Ryan and I tried but if we removed this variables means we will always create a doc store and a temporary dns for each verifiable document in the forms object in the template that was passed in, which is not what we are doing. Maybe you can help teach us how would we achieve the same result in the loop without creating multiple doc store and temp DNS for each verifiable document. We only want to create 1 doc store and temp DNS for multiple forms in the template.

Reply:
then dont do it that way. The current code look like
```
if (!verifiableDocumentDnsTxtName)
   verifiableDocumentDnsTxtName = await createTemporaryDnsWithDocumentStore();
```
which means, create a dns with(?) a document store and return a dns. Why returning the dns only ? It's inconsistent. I would suggest the following implementation


```
const documentStore = createDocumentStore();
const dnsTxt = createDnsTxt(documentStore);
```

where `createDocumentStore` create a document store once only (so store it as you already do): if document store exists, return it, otherwise create it and return it.
them for all the functions

Reply:
ok